twake/backend/node/src/services/tags/web/routes.ts

```diff
@@ -0,0 +1,47 @@
+import { FastifyInstance, FastifyPluginCallback } from "fastify";
+import { TagsController } from "./controllers";
+
+const tagsUrl = "/companies/:company_id/tags";
+
+const routes: FastifyPluginCallback = (fastify: FastifyInstance, options, next) => {
+  const tagsController = new TagsController();
+
+  fastify.route({
+    method: "GET",
+    url: `${tagsUrl}/:tag_id`,
+    preValidation: [fastify.authenticate],
+    handler: tagsController.get.bind(tagsController),
+  });
+
+  fastify.route({
+    method: "GET",
+    url: tagsUrl,
+    preValidation: [fastify.authenticate],
+    handler: tagsController.list.bind(tagsController),
+  });
+
+  fastify.route({
+    method: "POST",
+    url: tagsUrl,
+    preValidation: [fastify.authenticate],
+    handler: tagsController.save.bind(tagsController),
+  });
+
+  fastify.route({
+    method: "POST",
+    url: `${tagsUrl}/:tag_id`,
+    preValidation: [fastify.authenticate],
+    handler: tagsController.update.bind(tagsController),
+  });
+
+  fastify.route({
+    method: "DELETE",
+    url: `${tagsUrl}/:tag_id`,
+    preValidation: [fastify.authenticate],
+    handler: tagsController.delete.bind(tagsController),
+  });
+
```

Comment:
You will need to add a authentication check here for each route, you can add something in preValidation for that (a async function that returns true or false to go in this route or not).

You want to allow only people member of the company for getters, and administrators / owner of this company for setters.